modules/_labs/select/react/lib/Select.tsx

```diff
@@ -0,0 +1,342 @@
+  };
+
+  private toggleMenu = (show: boolean): void => {
+    if (show) {
+      this.setState({
+        focusedItemIndex: this.state.selectedItemIndex,
+        showingMenu: true,
+      });
+    } else {
+      this.setState({
+        focusedItemIndex: null,
+        showingMenu: false,
+      });
+    }
+  };
+
+  componentDidMount() {
+    const {children, value} = this.props;
+
+    const childrenArray = React.Children.toArray(children);
+
+    // if value exists, set state to that value...
+    if (value) {
+      const childIndex = this.indexByValue(value);
+
+      if (childIndex !== -1) {
+        this.setState({
+          label: childrenArray[childIndex].props.label,
+          selectedItemIndex: childIndex,
+        });
+        return;
+      }
+    }
+
+    // ... otherwise, set state to the first option
+    const firstOption = childrenArray[0];
+    // console.log('setting firstOption label:', firstOption.props.label);
+    this.setState({
+      label: firstOption.props.label,
+    });
+  }
+
+  handleSelectClick = (event: React.MouseEvent<HTMLInputElement>): void => {
+    this.toggleMenu(!this.state.showingMenu);
+  };
+
+  handleSelectBlur = (event: React.FocusEvent): void => {
+    this.toggleMenu(false);
+  };
+
+  handleOptionClick = (optionProps: SelectOptionProps): void => {
+    // disabled options cannot be clicked
+    if (optionProps.disabled) {
+      return;
+    }
+
+    const index = this.indexByValue(optionProps.value);
+    this.setState({
+      justSelectedItemIndex: index,
+      selectedItemIndex: index,
+    });
+
+    // offer visual feedback (flashing animation) briefly
+    // before dismissing menu; set new label when menu has been
+    // dismissed
+    setTimeout(() => {
+      this.setState({
+        justSelectedItemIndex: null,
+        label: optionProps.label,
+      });
+      this.toggleMenu(false);
+    }, dismissMenuTimeout);
+
+    if (this.inputRef && this.inputRef.current) {
+      const nativeInputValue = Object.getOwnPropertyDescriptor(
+        Object.getPrototypeOf(this.inputRef.current),
+        'value'
+      );
+      if (nativeInputValue && nativeInputValue.set) {
+        nativeInputValue.set.call(this.inputRef.current, optionProps.value);
+      }
+
+      let event: Event;
+      if (typeof Event === 'function') {
+        // modern browsers
+        event = new Event('input', {bubbles: true});
+      } else {
+        // IE 11
+        event = document.createEvent('Event');
+        event.initEvent('input', true);
+      }
+
+      this.inputRef.current.dispatchEvent(event);
+    }
+  };
+
+  handleKeyboardShortcuts = (event: React.KeyboardEvent): void => {
+    // console.log(event.key);
+
+    const children = React.Children.toArray(this.props.children);
+    const itemCount = children.length;
+    let isShortcut = false;
+    let nextFocusedIndex = 0;
+
+    switch (event.key) {
+      case 'ArrowUp':
```

Comment:
Please see this table for compatible key events for IE11: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key#Example